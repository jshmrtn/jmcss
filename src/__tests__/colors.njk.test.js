const colorsTemplate = require('template/colors.njk');

describe('colors.njk', () => {

  test('single color without transparency gets rendered', () => {
    let renderedTemplate;
    try {
      renderedTemplate = colorsTemplate.render({
        colors: [
          { name: 'test-color', r: 255, g: 255, b: 255, a: 1 },
        ],
      });
    } catch (error) {
      console.error(error);
    }
    expect(renderedTemplate).toEqual('$test-color: rgb(255, 255, 255);\n')
  });

  test('multiple colors without transparency get rendered', () => {
    let renderedTemplate;
    try {
      renderedTemplate = colorsTemplate.render({
        colors: [
          { name: 'test-color', r: 255, g: 255, b: 255, a: 1 },
          { name: 'test-color-two', r: 255, g: 255, b: 255, a: 1 },
        ],
      });
    } catch (error) {
      console.error(error);
    }
    expect(renderedTemplate).toMatchSnapshot();
    expect(renderedTemplate).toEqual(
      '$test-color: rgb(255, 255, 255);\n' +
      '$test-color-two: rgb(255, 255, 255);\n'
    );
  });

});
