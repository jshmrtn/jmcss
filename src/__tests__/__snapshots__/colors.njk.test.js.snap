// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colors.njk multiple colors without transparency get rendered 1`] = `
"$test-color: rgb(255, 255, 255);
$test-color-two: rgb(255, 255, 255);
"
`;
